bson/src/main/org/bson/codecs/pojo/PropertyReflectionUtils.java

```diff
@@ -60,24 +60,38 @@ static String toPropertyName(final Method method) {
     static PropertyMethods getPropertyMethods(final Class<?> clazz) {
         List<Method> setters = new ArrayList<Method>();
         List<Method> getters = new ArrayList<Method>();
-        for (Method method : clazz.getDeclaredMethods()) {
-            // Note that if you override a getter to provide a more specific return type, getting the declared methods
-            // on the subclass will return the overridden method as well as the method that was overridden from
-            // the super class. This original method is copied over into the subclass as a bridge method, so we're
-            // excluding them here to avoid multiple getters of the same property with different return types
-            if (isPublic(method.getModifiers()) && !method.isBridge()) {
-                if (isGetter(method)) {
-                    getters.add(method);
-                } else if (isSetter(method)) {
-                    // Setters are a bit more tricky - don't do anything fancy here
-                    setters.add(method);
+
+        // get all the default method from interface
+        for (Class<?> i : clazz.getInterfaces()) {
+            for (Method method : i.getDeclaredMethods()) {
+                if (method.isDefault()) {
+                    verifyAddMethodToList(method, getters, setters);
```

Comment:
Since we're using lists of getters and setters here, I think we have to be concerned with duplicates.  Consider the following scenario:

    interface WithDefault {
        default int getFoo() {
            return 42;
        }
    }

    static class WithDefaultImpl implements WithDefault {
        @Override
        public int getFoo() {
            return -42;
        }
    }

    public static void main(String[] args) {
        Class<WithDefaultImpl> clazz = WithDefaultImpl.class;

        for (Class<?> i : clazz.getInterfaces()) {
            for (Method method : i.getDeclaredMethods()) {
                if (method.isDefault()) {
                    System.out.println(method.getName());
                }
            }
        }

        for (Method method : clazz.getDeclaredMethods()) {
            System.out.println(method.getName());
        }
    }

Because `WithDefaultImpl` overrides the default interface method, this will print `getFoo` twice.   So in this path, it would result in `getFoo` being added to the `getters` list twice.  I'm not sure what the ramifications of that are, but it's probably not good (especially for the setter, which it may result in a duplicate field being added to the BSON document.

So do we have to check for duplicates? Perhaps the caller of this method handles duplicates so it's not a problem.  But let's add a test so that we understand the behavior.

I'm honestly not clear what the use case is for default getters and setters that are not overridden by the implementing class.

Reply:
I have added a test that overrides a default interface method. No changes were needed in PropertyReflectionUtils.

Reply:
Although the `InterfaceWithOverrideDefaultMethodModelImpl` round trips ok - we should ensure we add the default interface methods last.  That way the actual method / type is used by the PojoBuilderHelper.  See: https://github.com/jstewart-mongo/mongo-java-driver/blob/j3148/bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java#L84-L102

Reply:
I now add the default interface methods last.
Evergreen patch: https://evergreen.mongodb.com/version/5db07dbec9ec44772065505d

Reply:
Can we write a test that proves this is having the desired effect?

Reply:
I added two tests to PojoRoundTripTest.java to test adding default interface methods and overriding a default method.

Reply:
Right, but there was no test change associated with patch 3, so I'm not clear what it actually accomplishes.  Is there a user-visible behavioral change and if so, can we write a test to demonstrate it.

Reply:
I have looked into the changes Ross pointed to for setters, but I do not see how to write a test to demonstrate a user-visible change. Adding default interface methods last just guarantees that the actual method will be added.